extensions/csv/src/main/java/com/hazelcast/jet/csv/impl/CsvReadFileFnProvider.java

```diff
@@ -68,4 +105,17 @@
     public String format() {
         return CsvFileFormat.FORMAT_CSV;
     }
+
+    private static int[] createSimpleFieldMap(List<String> fieldList, String[] actualHeader) {
+        int[] res = new int[fieldList.size()];
+        Arrays.fill(res, -1);
+        for (int i = 0; i < actualHeader.length; i++) {
+            int index = fieldList.indexOf(actualHeader[i]);
+            // if the header is present in the file and we didn't encounter it yet, store its index
+            if (index >= 0 && res[index] == -1) {
```

Comment:
Is `res[index] == -1` condition needed?

Reply:
Yes, that's to ensure that if a duplicate header is found, the first one is used, not the last one. We don't overwrite index for a column that already has an index.